xarray/backends/common.py

```diff
@@ -264,19 +264,22 @@ def add(self, source, target):
         else:
             target[...] = source
 
-    def sync(self):
+    def sync(self, compute=True):
         if self.sources:
             import dask.array as da
-            da.store(self.sources, self.targets, lock=self.lock)
+            delayed = da.store(self.sources, self.targets, lock=self.lock,
+                               compute=compute, flush=True)
```

Comment:
I guess this is where the magic happens. You are just exposing `dask.array.store(... compute=False)` as an option from xarray.

Reply:
@jakirkham has been working on `da.store` a bit recently.  He may have thoughts here.

Reply:
Not sure that `flush` does anything actually. Generic `**kwargs` get forwarded onto `compute`/`persist` depending. 

FWIW my typical usage is `compute=True, return_stored=True`. This is pretty nice as one gets back a Dask Array with the ability to load in data that has been stored to disk. Pretty nice for inspecting the result as it computes. Can also be tossed into a Dask progress bar to get details about how much has completed.